src/lib.rs

```diff
@@ -144,6 +145,68 @@ pub fn demangle(mut s: &str) -> Demangle {
     }
 }
 
+#[cfg(feature = "std")]
+fn demangle_line(line: &str, include_hash: bool) -> std::borrow::Cow<str> {
```

Comment:
If the goal of this function is primarily speed, would it perhaps be beneficial to take a `&str` and `Write` and avoid returning a string?

Reply:
At least in my profiling that ended up being a little slower - I suspect that LLVM is better able to detect that io/fmt::Write to String can't fail (at least not via io::Error), though I'm definitely not confident that's the cause. 

In any case at least for the rustfilt case this amounts to just a few percent (or less) of the overall runtime, the vast majority is spent in the demangling routines. I think there's room to optimize there, and we can change the implementation here if this becomes more significant - this function is intentionally not public API.